tls-client/main.cpp

```diff
@@ -210,29 +207,39 @@ class HelloHTTPS {
         ret = _tcpsocket->connect(_domain, _port);
         if (ret != NSAPI_ERROR_OK) {
             mbedtls_printf("Failed to connect\r\n");
-            onError(_tcpsocket, -1);
+            printf("MBED: Socket Error: %d\r\n", ret);
+            _tcpsocket->close();
```

Comment:
For the first comment. Having a status print in ```main()``` deviates from the issue this PR addresses. Secondly that would make this example look more like a test case. Success or failure is apparent by looking the progress in the logs.

Second comment: putting ```_tcpsocket->close()``` in destructor was a good idea. But @yanesca wanted to maintain coherence by also moving ```_tcpsocket->connect()``` in the constructor. That I don't agree with that as connection failure will result in an object that isn't in the expected state.